torch/_tensor.py

```diff
@@ -585,8 +585,6 @@ def __iter__(self):
         # (e.g., if you zip(*hiddens), the eager map will force all the
         # indexes of hiddens[0] before hiddens[1], while the generator
         # map will interleave them.)
-        if has_torch_function_unary(self):
-            return handle_torch_function(Tensor.__iter__, (self,), self)
```

Comment:
Probably good to have a comment here saying that torch function handling is intentionally omitted

Reply:
Done in https://github.com/pytorch/pytorch/pull/54702/commits/e6693244022b495a42020c4ee3b00424edbca8e7.